From 7d3baf86e919f067197a12dc3267a4c16268b6e5 Mon Sep 17 00:00:00 2001
From: Srinidhi Kasagar <srinidhi.kasagar@intel.com>
Date: Fri, 13 Feb 2015 12:02:42 +0530
Subject: [PATCH] intel_idle: Disable core and module c6 auto-demotion for
 cherrytrail

Disabling core and module C6 auto-demotion helps even cherrytrail.
Let's make the flag generic and apply to this platform as well.

Tracked-On: https://jira01.devtools.intel.com/browse/GMIN-2952
Signed-off-by: Srinidhi Kasagar <srinidhi.kasagar@intel.com>

Change-Id: I841095afaae082bea9eb8ec96acb157f929c5a26
(cherry picked from commit 462b7dc8123d66952b0588a34a12871fec037364)

Signed-off-by: Nagaraj S <sivasankaranx.nagaraj@intel.com>
---
 drivers/idle/intel_idle.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index 3896e8e..1291e04 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -89,7 +89,7 @@ struct idle_cpu {
 	 * Indicate which enable bits to clear here.
 	 */
 	unsigned long auto_demotion_disable_flags;
-	bool byt_auto_demotion_disable_flag;
+	bool auto_demotion_disable_flag;
 	bool disable_promotion_to_c1e;
 };
 
@@ -544,12 +544,13 @@ static const struct idle_cpu idle_cpu_snb = {
 static const struct idle_cpu idle_cpu_byt = {
 	.state_table = byt_cstates,
 	.disable_promotion_to_c1e = true,
-	.byt_auto_demotion_disable_flag = true,
+	.auto_demotion_disable_flag = true,
 };
 
 static const struct idle_cpu idle_cpu_cht = {
 	.state_table = cht_cstates,
 	.disable_promotion_to_c1e = true,
+	.auto_demotion_disable_flag = true,
 };
 
 static const struct idle_cpu idle_cpu_ivb = {
@@ -710,7 +711,7 @@ static int __init intel_idle_cpuidle_driver_init(void)
 	if (icpu->auto_demotion_disable_flags)
 		on_each_cpu(auto_demotion_disable, NULL, 1);
 
-	if (icpu->byt_auto_demotion_disable_flag) {
+	if (icpu->auto_demotion_disable_flag) {
 		wrmsrl(MSR_CC6_DEMOTION_POLICY_CONFIG, 0);
 		wrmsrl(MSR_MC6_DEMOTION_POLICY_CONFIG, 0);
 	}
-- 
1.7.9.5

